src/outputs/parthenon_hdf5.cpp

```diff
@@ -825,7 +829,30 @@ void PHDF5Output::WriteOutputFileImpl(Mesh *pm, ParameterInput *pin, SimTime *tm
     const hsize_t nx5 = vinfo.nx5;
     const hsize_t nx4 = vinfo.nx4;
 
-    local_count[1] = global_count[1] = vlen;
+    local_count[1] = global_count[1] = nx6;
+    local_count[2] = global_count[2] = nx5;
+    local_count[3] = global_count[3] = nx4;
+
+    std::vector<hsize_t> alldims({nx6, nx5, nx4, static_cast<hsize_t>(nx3),
+                                  static_cast<hsize_t>(nx2), static_cast<hsize_t>(nx1)});
+
+    int ndim = -1;
+    if (vinfo.where == MetadataFlag(Metadata::Cell)) {
+      ndim = 3 + vinfo.ndim + 1;
+      for (int i = 0; i < vinfo.ndim; i++) {
+        local_count[1 + i] = global_count[1 + i] = alldims[3 - vinfo.ndim + i];
+      }
+      local_count[vinfo.ndim + 1] = global_count[vinfo.ndim + 1] = nx3;
+      local_count[vinfo.ndim + 2] = global_count[vinfo.ndim + 2] = nx2;
+      local_count[vinfo.ndim + 3] = global_count[vinfo.ndim + 3] = nx1;
+    } else if (vinfo.where == MetadataFlag(Metadata::None)) {
+      ndim = vinfo.ndim + 1;
+      for (int i = 0; i < vinfo.ndim; i++) {
+        local_count[1 + i] = global_count[1 + i] = alldims[6 - vinfo.ndim + i];
+      }
```

Comment:
@Yurlungur here is where we need to know both `shape` and whether the variable is e.g. tied to cell centers (and thus whether its last three entries in `*_count` will be `nx3`, `nx2`, `nx1` or not)

Reply:
I could be missing something but I think the issue here basically shows up because we want to suppress the unused dimensions in the ParArray e.g. `nx4`, `nx5`, `nx6` for a scalar field when generating the HDF5 dataset. But since these are always >= 1, we can't just look at the `nx*` to determine the actual shape of the desired HDF5 dataset.